cmd/auth/routes.go

```diff
@@ -7,19 +7,34 @@ import (
 
 	configsource "github.com/skygeario/skygear-server/pkg/auth/config/source"
 	"github.com/skygeario/skygear-server/pkg/auth/dependency/webapp"
+	"github.com/skygeario/skygear-server/pkg/auth/handler/internalserver"
 	oauthhandler "github.com/skygeario/skygear-server/pkg/auth/handler/oauth"
-	"github.com/skygeario/skygear-server/pkg/auth/handler/session"
 	webapphandler "github.com/skygeario/skygear-server/pkg/auth/handler/webapp"
 	"github.com/skygeario/skygear-server/pkg/core/sentry"
-	"github.com/skygeario/skygear-server/pkg/core/server"
 	"github.com/skygeario/skygear-server/pkg/deps"
+	"github.com/skygeario/skygear-server/pkg/httputil"
 )
 
 func NewRouter(p *deps.RootProvider) *mux.Router {
-	rootRouter := mux.NewRouter()
-	rootRouter.Use(p.RootMiddleware(newSentryMiddlewareFactory(sentry.DefaultClient.Hub)))
-	rootRouter.Use(p.RootMiddleware(newRecoverMiddleware))
-	return rootRouter
+	r := mux.NewRouter()
+	r.Use(p.RootMiddleware(newSentryMiddlewareFactory(sentry.DefaultClient.Hub)))
+	r.Use(p.RootMiddleware(newRecoverMiddleware))
+	r.HandleFunc("/healthz", httputil.HealthCheckHandler)
+	return r
+}
+
+func setupInternalRoutes(p *deps.RootProvider, configSource configsource.Source) *mux.Router {
+	r := NewRouter(p)
+
+	r.Use((&deps.RequestMiddleware{
```

Comment:
This would make health check fail if app is not found. We should create a sub-router for the real routes.